### PR TITLE
Adding pipeline to push ecr images

### PR DESCRIPTION
**Pull request by MbolotSuse**

### CODE REVIEW

**Comment by MbolotSuse** [comment]
Related to rancher/rancher#37757.

## Overview
The CSP adapter is going to be published as an aws marketplace offering. Because of this, the image for the adapter must be published in ECR. This PR attempts to add the pipeline stages necessary to push to ecr for amd and arm images.

## Changes
- Added a stage to the linux-amd64 pipeline to push to the ECR repo. This stage uses the ecr plugin. This stage only runs on tag events that don't include a dash (for example it will run on `1.2.3` but will not on `1.2.3.-rc1`).
- Added a stage to the linux-arm64 pipeline to push to  the ECR repo. This stage uses the ecr plugin. This stage only runs on tag events that don't include a dash (for example it will run on `1.2.3` but will not on `1.2.3.-rc1`).
- Added a stage to the manifest pipeline to push the manifest to the ecr repo. This stage does not use the ecr plugin (which doesn't appear to support pushing manifests) or the manifest plugin (which does not appear to support ecr). Instead, it sets the secret key/access key as env vars (in line with [aws docs](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html))

## FAQ
- Why do we not push on rc tags like we do to dockerhub?
  - [AWS docs](https://docs.aws.amazon.com/marketplace/latest/userguide/container-product-policies.html) for container products specify that no beta products can go in ECR.
- Why do we use a more custom pipeline for the manifest push phase?
  - There did not appear to be a more simple solution for pushing a manifest to ecr. No available plugin seemed to support both ECR and manifests

File: .drone.yml [hunk 1]
```diff
@@ -63,6 +63,26 @@ steps:
     event:
     - pull_request
 
+- name: ecr-publish
+  image: plugins/ecr
+  environment:
+    AWS_REGISTRY_ID:
```
Comment (oxr463) [inline .drone.yml]: Does this need to be an environment variable too?

**Review by oxr463 — APPROVED** [review]

**Comment by oxr463** [comment]
Also, we need to make sure those secrets are added in Drone.

File: .drone.yml (hunk 1 above)
Comment (MbolotSuse) [reply .drone.yml]: That's a good point. I use it for the registry block, maybe there's a way I can reference the secret there directly.

Comment (oxr463) [reply .drone.yml]: Yeah, that's what I would do. It doesn't matter though, more of a nitpick lol

**Review by oxr463 — APPROVED** [review]

**Review by nettlingham — APPROVED** [review]

**Pull request merged**